dialer.go

```diff
@@ -170,6 +170,20 @@ func NewDialer(ctx context.Context, opts ...Option) (*Dialer, error) {
 	return d, nil
 }
 
+func verifyIAMAuthNConfig(version string, wantsIAMAuthN bool) error {
+	// Don't check engine version if IAM AuthN isn't requested.
+	if !wantsIAMAuthN {
+		return nil
+	}
+	v := strings.ToLower(version)
```

Comment:
This is probably unnecessary because all the types are currently capitalized: https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/SqlDatabaseVersion

so we can just use `strings.HasPrefix(v, "POSTGRES")`. 

If we are going to sanitize I'd rather be to caps since it should be a noop